### PR TITLE
Add profile functionality for admin and supervisor roles

### PR DESCRIPTION
**Pull request by Magi1610**
This pull request introduces user profile functionality for both admin and supervisor roles, with support for handling department-specific profile images. The following changes were made:

.gitignore: Updated to reflect new file exclusions.

HomeController.php: Modified to redirect users based on their roles (admin or supervisor) and added support for displaying the correct profile information.

ProfileController.php: Added to handle profile actions such as editing profile pictures, logos, and passwords.

User.php: Updated relationships to handle media (profile and department pictures) and added necessary helper methods.

adminlte.php & app.php: Configurations adjusted to incorporate user profile and department image management.

2024_09_25_053151_create_media_table.php: Added down function to the migration, enabling rollback of the media table.

RoleSeeder.php: Updated to seed roles for admin and supervisor users.

profile.css: New CSS file for styling profile pages.

home.blade.php: Adjusted to display user-specific information and department logos.

Profile views (editImage, editLogo, editPassword, index.blade.php): Added new views to manage user profile updates, department logos, and password changes.

web.php: Updated routes to handle profile management.

AdminLTE Navbar user menu (menu-item-dropdown-user-menu.blade.php): Customized to display the department name and profile picture for supervisors, and the username for admins.

### CODE REVIEW

File: app/Http/Controllers/ProfileController.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+        if ($newPassword !== $confirmPassword) {
+            return redirect()->back()->with('error', 'La nueva contraseña y la confirmación no coinciden.');
+        }
+
+        $user->password = Hash::make($newPassword);
+        $user->save();
+
+        return redirect()->route('profiles.index')->with('success', 'Contraseña actualizada correctamente.');
+    }
```
Comment (And-Y21) [inline app/Http/Controllers/ProfileController.php:68]: Falta un espacio

File: app/Http/Controllers/ProfileController.php [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        return redirect()->route('profiles.index')->with('success', 'Perfil actualizado correctamente.');
+    }
+
+    public function updateImage(Request $request)
+    {
+        $request->validate([
+            'profileImage' => 'required|image|mimes:jpeg,png,jpg,gif|max:2048',
+        ]);
+
+        $user = User::find(Auth::id());
```
Comment (josejulianmtnez) [inline app/Http/Controllers/ProfileController.php:38]: Homologa estas variables en unos lados los llamas $user y en otros $authUser

File: resources/views/home.blade.php [hunk 3]
```diff
@@ -8,6 +8,25 @@
...
+                <div class="row align-items-center">
+                    <div class="col-md-2 text-center">
+                        @if ($authUser->getFirstMediaUrl('userGallery'))
+                            <img src="{{ $authUser->getFirstMediaUrl('userGallery') }}"
+                                alt="Foto de {{ $authUser->name }}">
+                        @else
+                            <img src="{{ asset('img/userDefault.png') }}">
+                        @endif
+                    </div>
+
```
Comment (And-Y21) [inline resources/views/home.blade.php]: Hay un espacio

File: database/seeders/RoleSeeder.php [hunk 4]
```diff
@@ -26,7 +26,7 @@ public function run(): void
         Permission::create(['name' => 'category', 'description' => 'Visualizar Categorias'])->assignRole($role2);
         Permission::create(['name' => 'students', 'description' => 'Visualizar Estudiantes'])->assignRole($role2);
         Permission::create(['name' => 'materials', 'description' => 'Visualizar Materiales'])->assignRole($role2);
-        Permission::create(['name' => 'departments', 'description' => 'Visualizar Departamentos'])->assignRole($role2);
+        Permission::create(['name' => 'departments', 'description' => 'Visualizar Departamentos'])->assignRole($role1);
```
Comment (josejulianmtnez) [inline database/seeders/RoleSeeder.php]: Mejoren el nombre de sus variables asignadas a cada rol
No los llamen como role1 o role2, etc

File: resources/views/home.blade.php [hunk 5]
```diff
@@ -99,6 +118,7 @@
 @stop
 
 @section('css')
+    <link rel="stylesheet" href="{{ asset('css/profiles/profile.css') }}">
     <style>
```
Comment (And-Y21) [inline resources/views/home.blade.php]: ¿Por qué hay una etiqueta de style abajo y arriba referencias a un link de style?

File: routes/web.php [hunk 6]
```diff
@@ -31,6 +32,12 @@
 Route::middleware(['auth'])->group(function () {
     Route::get('/home', [App\Http\Controllers\HomeController::class, 'index'])->name('home');
 
+    Route::get('/profiles', [ProfileController::class, 'index'])->name('profiles.index');
+    Route::put('/profiles/update', [ProfileController::class, 'profileUpdate'])->name('profiles.update');
+    Route::post('/profiles/updateImage', [ProfileController::class, 'updateImage'])->name('profiles.updateImage');
+    Route::post('/profiles/updatePicture', [ProfileController::class, 'updatePicture'])->name('profiles.updatePicture');
+    Route::put('/profiles/editPassword', [ProfileController::class, 'updatePassword'])->name('profiles.updatePassword');
+
```
Comment (And-Y21) [inline routes/web.php]: Línea de mas

File: vendor/jeroennoten/laravel-adminlte/resources/views/partials/navbar/menu-item-dropdown-user-menu.blade.php [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+            <form id="logout-form" action="{{ $logout_url }}" method="POST" style="display: none;">
+                @if(config('adminlte.logout_method'))
+                    {{ method_field(config('adminlte.logout_method')) }}
+                @endif
+                {{ csrf_field() }}
+            </form>
+        </li>
+
+    </ul>
+
```
Comment (josejulianmtnez) [inline vendor/jeroennoten/laravel-adminlte/resources/views/partials/navbar/menu-item-dropdown-user-menu.blade.php:94]: Cuiden el identado de esta parte

File: app/Models/User.php [hunk 8]
```diff
@@ -51,4 +51,31 @@ public function department()
...
+            $departmentImage = $this->getFirstMediaUrl('departmentGallery');
+            return $departmentImage ?: url('img/logo.png');
+        }
+    }
+
+    public function adminlte_desc()
+    {
+        $role = $this->getRoleNames()->first();
+        return $role ?? 'Unknown Role';
+
```
Comment (And-Y21) [inline app/Models/User.php]: Salto de línea innecesario

File: vendor/jeroennoten/laravel-adminlte/resources/views/partials/navbar/menu-item-dropdown-user-menu.blade.php (hunk 7 above)
Comment (josejulianmtnez) [inline vendor/jeroennoten/laravel-adminlte/resources/views/partials/navbar/menu-item-dropdown-user-menu.blade.php:94]: Cuiden el identado de esta parte

**Pull request merged**